datapy/Table.py

```diff
@@ -0,0 +1,178 @@
+
+        Returns:
+            bool: Was the table created?
+        """
+        headers = [h.replace("::", " ") for h in headers.split(",")]
+        create_stmt = f"create table if not exists {self.table_name} ( {', '.join(headers)} )"
+        c = conn.cursor()
+        try:
+            c.execute(create_stmt)
+            return True
+        except:
+            return False
+
+    def load_table(self, conn):
+        """ loads tables from a previous db dump
+        Args:
+            self.table_root (str): path to table
+            conn (sql.Connection): connection to the database
+        """
+        c = conn.cursor()
+        
+    def ingest_csv(self, conn):
+        """ Read a properly formatted .csv file and stash it in a sqlite table
+        Args:
+            self.input_data (str): path to the .csv file
+            conn (sql.Connection): connection to the database
+        Returns:
+            int: the number of table rows inserted
+        """
+        with open(self.input_data) as f, conn:
+            headers = f.readline().strip()
+            created = self.create_table(conn, headers)
+            if created:
+                rows_inserted = self.insert_rows(conn, headers, f)
+            try:
+                conn.commit()
+                return rows_inserted
+            except:
+                sys.exit("No rows were inserted, error")
+
+    def insert_rows(self, conn, headers, file_conn):
+        """ Insert rows from a csv file connection into a database table
+        Args:
+            self.table_name (str): name of the table to insert records into
+            conn (sql.Connection): connection to the database
+            headers (str): table headers in string format -
+                           "field::type,field::type"
+            file_conn (TextIOWrapper): file connection
+
+        Returns:
+            int: number of records inserted
+        """
+        types = [h.split("::")[1] for h in headers.split(",")]
+        headers = [h.split("::")[0] for h in headers.split(",")]
+        reader = csv.DictReader(file_conn, fieldnames=headers)
+        rows_inserted = 0
+        c = conn.cursor()
+        for row in reader:
+            insert_stmt = (
+                f"insert into {self.table_name} ({','.join(headers)}) "
+                f"values ({','.join(['?']*len(headers))});"
+                )
+            c.execute(insert_stmt, tuple(row.values()))
+            rows_inserted += 1
+        return rows_inserted
+
+    def select_all_records(self, conn):
+        """ Return all records from a named table
+        Args:
+            self.table_name (str): name of the table to fetch from
+            conn (sql.Connection): connection to the database
+
+        Returns:
+            List[Tuple[Any]]: A list of returned records
+        """
+        with conn:
+            c = conn.cursor()
+            print(self.table_name)
+            try:
+                c.execute(f"select * from {self.table_name}")
```

Comment:
I would like this to call another function that allows for arguments so that we aren't implementing hardcoded selects in multiple places.